lib/matplotlib/image.py

```diff
@@ -695,8 +695,8 @@ def set_data(self, A):
 
         if (self._A.dtype != np.uint8 and
                 not np.can_cast(self._A.dtype, float, "same_kind")):
-            raise TypeError("Image data of dtype {} cannot be converted to "
-                            "float".format(self._A.dtype))
+            raise TypeError("Image data of dtype {} cannot be converted to 'float'"
```

Comment:
perhaps just make an f-string, and remove the inner quotes.  The error in the old code was the spurious carriage return, not that we were trying to put float in quotes...

Reply:
The carriage return is not an error (the strings get implicitly concatenated *before* .format() is called).  I don't actually think there's anything to change here...

Reply:
Ah, I didn't realize they concatenated before `.format()`.  Agree there is nothing to change here.